pkg/cmd/pulumi/plugin_install.go

```diff
@@ -137,13 +137,11 @@ func (cmd *pluginInstallCmd) Run(ctx context.Context, args []string) error {
 			}
 		}
 
-		pluginSpec := workspace.PluginSpec{
-			Kind:              apitype.PluginKind(args[0]),
-			Name:              args[1],
-			Version:           version,
-			PluginDownloadURL: cmd.serverURL, // If empty, will use default plugin source.
-			Checksums:         checksums,
+		pluginSpec, err := workspace.NewPluginSpec(args[1], apitype.PluginKind(args[0]), cmd.serverURL, checksums)
+		if err != nil {
+			return err
 		}
+		pluginSpec.Version = version
```

Comment:
I'm not sure about this. 
A) Feels like serverURL and treating args[1] as a source could conflict. 
B) Version and version in args[1] could definitely conflict

I feel like "the right thing" is NewPluginSpec needs to parse into a spec, then we need to check here, "did NewPluginSpec set version, if yes then if the user also set --version that's a conflict error, same for serverURL". 

Doing that probably either means moving all that logic into NewPluginSpec (so it should take a version argument as well). Or just reducing newPluginSepc to just take the source string and kind.
Given checksums needs setting as well, the former is probably nicer.

Reply:
Yeah, we already do the conflict check for the pluginDownloadURL in `NewPluginSpec`, so I think it makes sense to do the same with `version`.  Updated the PR to do that now.